here's my error logs.

```
platform/windows/godot_win.cpp: In function 'int widechar_main(int, wchar_t**)':
platform/windows/godot_win.cpp:128:13: error: cannot declare variable 'os' to be of abstract type 'OS_Windows'
  OS_Windows os(NULL);
             ^
In file included from platform/windows/godot_win.cpp:29:0:
platform/windows/os_windows.h:62:7: note:   because the following virtual functions are pure within 'OS_Windows':
 class OS_Windows : public OS {
       ^
In file included from platform/windows/os_windows.h:35:0,
                 from platform/windows/godot_win.cpp:29:
core/os/os.h:255:23: note: 	virtual OS::TimeZoneInfo OS::get_time_zone_info() const
  virtual TimeZoneInfo get_time_zone_info() const=0;
                       ^
x86_64-w64-mingw32-g++ -o main/main.windows.opt.tools.64.o -c -O2 -DDEBUG_ENABLED -DFREETYPE_ENABLED -DWINDOWS_ENABLED -mwindows -DGLES2_ENABLED -DGLEW_ENABLED -DRTAUDIO_ENABLED -DMINGW_ENABLED -DMUSEPACK_ENABLED -DOPENSSL_ENABLED -DSQUISH_ENABLED -DVORBIS_ENABLED -DPNG_ENABLED -DDDS_ENABLED -DPVR_ENABLED -DJPG_ENABLED -DWEBP_ENABLED -DSPEEX_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DETC1_ENABLED -DOPENSSL_NO_ASM -DOPENSSL_THREADS -DL_ENDIAN -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/windows -Itools/freetype -Itools/freetype/freetype/include -Idrivers/builtin_openssl2 -Idrivers/builtin_zlib/zlib -Imain/vorbis -Idrivers/builtin_openssl2/crypto -Idrivers/builtin_openssl2/openssl -Idrivers/builtin_openssl2/crypto/evp -Idrivers/builtin_openssl2/crypto/asn1 -Idrivers/builtin_openssl2/crypto/modes main/main.cpp
scons: *** [platform/windows/godot_win.windows.opt.tools.64.o] Error 1
platform/windows/os_windows.cpp:1863:51: error: no 'OS::TimeZoneInfo OS_Windows::get_time_zone_info() const' member function declared in class 'OS_Windows'
 OS::TimeZoneInfo OS_Windows::get_time_zone_info() const {
                                                   ^
scons: *** [platform/windows/os_windows.windows.opt.tools.64.o] Error 1
scons: building terminated because of errors.
```

--- a/platform/windows/os_windows.cpp
+++ b/platform/windows/os_windows.cpp
@@ -1829,22 +1829,7 @@ String OS_Windows::get_name() {
 	return "Windows";
 }
 
-OS::Date OS_Windows::get_date() const {
-
-	SYSTEMTIME systemtime;
-	if (local)
-		GetSystemTime(&systemtime);
-	else
-		GetLocalTime(&systemtime);
-	Date date;
-	date.day=systemtime.wDay;
-	date.month=Month(systemtime.wMonth);
-	date.weekday=Weekday(systemtime.wDayOfWeek);
-	date.year=systemtime.wYear;
-	date.dst=false;
-	return date;
-}
-OS::Time OS_Windows::get_time(bool utc) const {
+OS::Date OS_Windows::get_date(bool utc) const {
 
 	SYSTEMTIME systemtime;
 	if (utc)
@@ -1852,11 +1837,13 @@ OS::Time OS_Windows::get_time(bool utc) const {
 	else
 		GetLocalTime(&systemtime);
 
-	Time time;
-	time.hour=systemtime.wHour;
-	time.min=systemtime.wMinute;
-	time.sec=systemtime.wSecond;
-	return time;
+	Date date;
+	date.day=systemtime.wDay;
+	date.month=Month(systemtime.wMonth);
+	date.weekday=Weekday(systemtime.wDayOfWeek);
+	date.year=systemtime.wYear;
+	date.dst=false;
+	return date;
 }
 OS::Time OS_Windows::get_time(bool utc) const {
 
@@ -1889,7 +1876,7 @@ OS::TimeZoneInfo OS_Windows::get_time_zone_info() const {
 	return ret;
 }
 
-uint64_t OS_Windows::get_unix_time(bool local) const {
+uint64_t OS_Windows::get_unix_time() const {
 
 	FILETIME ft;
 	SYSTEMTIME st;

--- a/platform/windows/os_windows.h
+++ b/platform/windows/os_windows.h
@@ -261,6 +261,7 @@ public:
 	
 	virtual Date get_date(bool utc) const;
 	virtual Time get_time(bool utc) const;
+	virtual TimeZoneInfo get_time_zone_info() const;
 	virtual uint64_t get_unix_time() const;
 
 	virtual bool can_draw() const;

--- a/platform/winrt/os_winrt.cpp
+++ b/platform/winrt/os_winrt.cpp
@@ -442,10 +442,14 @@ String OSWinrt::get_name() {
 	return "WinRT";
 }
 
-OS::Date OSWinrt::get_date() const {
+OS::Date OSWinrt::get_date(bool utc) const {
 
 	SYSTEMTIME systemtime;
-	GetLocalTime(&systemtime);
+	if (utc)
+		GetSystemTime(&systemtime);
+	else
+		GetLocalTime(&systemtime);
+
 	Date date;
 	date.day=systemtime.wDay;
 	date.month=Month(systemtime.wMonth);
@@ -485,14 +489,11 @@ OS::TimeZoneInfo OS_Windows::get_time_zone_info() const {
 	return ret;
 }
 
-uint64_t OSWinrt::get_unix_time(bool utc) const {
+uint64_t OSWinrt::get_unix_time() const {
 
 	FILETIME ft;
 	SYSTEMTIME st;
-	if (utc)
-		GetSystemTime(&systemtime);
-	else
-		GetLocalTime(&systemtime);
+	GetSystemTime(&systemtime);
 	SystemTimeToFileTime(&st, &ft);
 
 	SYSTEMTIME ep;

--- a/platform/winrt/os_winrt.h
+++ b/platform/winrt/os_winrt.h
@@ -200,6 +200,7 @@ public:
 	
 	virtual Date get_date(bool utc) const;
 	virtual Time get_time(bool utc) const;
+	virtual TimeZoneInfo get_time_zone_info() const;
 	virtual uint64_t get_unix_time() const;
 
 	virtual bool can_draw() const;

